### PR TITLE
WINDUPRULE-660 Added Spring Boot Actuator ruleset

### PR DESCRIPTION
**Pull request by rmarting**
First implementation to cover the [SB2Quarkus - Spring Boot Actuator](https://issues.redhat.com/browse/WINDUPRULE-660) ruleset.

Includes a simple rule to identify when Spring Boot Actuator dependency is included.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by jonathanvila** [comment]
Hi @rmarting thanks for creating this PR :+1: 

In order to make it perfect I would suggest to create a test for the rule as you can see in this example : https://issues.redhat.com/browse/WINDUPRULE-476

File: rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    </metadata>
+    <rules>
+        <rule id="springboot-actuator-to-quarkus-0100">
+            <when>
+                <dependency groupId="org.springframework.boot" artifactId="spring-boot-starter-actuator"/>
+            </when>
+            <perform>
+                <hint title="Replace the Spring Boot Actuator dependency with Quarkus Smallrye Health extension" effort="5" category-id="mandatory">
+                    <message>
+                        Replace the Spring Boot Actuator dependency with Quarkus Smallrye Health extension
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml:23]: If you want to create a new line between the two sentences, you should either add a blank line or add 2 spaces at the end of the first line.

File: rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="springboot-metrics-to-quarkus" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml:2]: Is the id correct ? "springboot-metrics-to-quarkus"

File: rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <sourceTechnology id="springboot"/>
+        <targetTechnology id="quarkus"/>
+    </metadata>
+    <rules>
+        <rule id="springboot-actuator-to-quarkus-0100">
+            <when>
+                <dependency groupId="org.springframework.boot" artifactId="spring-boot-starter-actuator"/>
+            </when>
+            <perform>
+                <hint title="Replace the Spring Boot Actuator dependency with Quarkus Smallrye Health extension" effort="5" category-id="mandatory">
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml:21]: If it's only a matter of changing a dependency, I would say that effort=1 as you could find here : https://github.com/windup/windup-rulesets/pull/431/files

File: rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml (hunk 1 above)
Comment (rmarting) [reply rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml]: Done! I did not know that "magic".

File: rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml (hunk 2 above)
Comment (rmarting) [reply rules-reviewed/quarkus/springboot/springboot-actuactor-to-quarkus.windup.xml]: Typo fixed!

**Comment by rmarting** [comment]
Added a test case to verify the rule. It is the first time that define a ruleset with the latest version of windup (my first rules were on windup 2 and 3 :smiley:).
Any other feedback?

**Comment by PhilipCattanach** [comment]
Tested successfully and code reviewed by myself and Marco Rizzi.

**Pull request merged**